### DISCUSSION TITLE
Triaging and Handling Issues

### DISCUSSION THREAD

**Discussion by adiati98** [petermsouzajr/qa-shadow-report#49] [General]
Hi @petermsouzajr,

I'm writing this on discussion instead of an issue to gain your thoughts. 😊

Looking at the docs, at the moment, there is no information about triaging issues and how contributors can work on an issue.

Here are some scenarios:

- In the future, you might encounter receiving issues from contributors that you don't actually want to apply to your project. But the contributors have created PRs for those issues.
- You might receive PRs that offer changes for the same issue from multiple contributors. Remember that contributors have put their times and efforts in working on these PRs. In this case, which one do you want to accept? Or would it make you confuse?

Having clear information about triaging issues and how contributors can get assigned and working on an issue can help in preventing double PRs, time wasting, and awkwardness when maintainers should close a duplicate PRs.

There are several ways in preventing these scenarios that you can put in the docs:

- Add `needs triage` label to issues from external contributors.

   As long as this label hasn't been removed by maintainers, this issue is not available for contribution. Clearly state this in the docs.
   You can also automate this by using [GitHub Action to add issue labels](https://docs.github.com/en/actions/managing-issues-and-pull-requests/adding-labels-to-issues).

- Assign contributors

   Assigning contributors to an issue is one way to prevent duplicate PRs. If an issue is assigned to someone, it's no longer available for anyone else.

   You can:
   - state in the docs something like:

     > Please comment on the issue if you're interested working on it. Wait until a maintainer assigns the issue to you before you start working on changes. **We will not reveice and approve a PR without assigned issue**.

   - Or, you can let contributors assign the issue to themselves.

     By default, on GitHub, only maintaners can assign issue to contributors. But there is a [GitHub Action](https://github.com/marketplace/actions/contributor-takes-action) that you can use for this purpose if you would like.

**Comment by petermsouzajr** [comment]
thank you, staying on the theme of 'removing ambiguity'. Great advice, i will consider some solutions and get them in today.

**Comment by petermsouzajr** [comment]
Thank you for the great suggestions, I added the github actions to: automatically add the `needs triage` label to new issues.  Ability for contributors to comment on an available issue so that they are automatically assigned.

added more details to the CONTRIBUTING document to describe the process of claiming an issue.

Hopefully these new changes will prevent duplicate work or code conflicts.
